decoder/src/lib.rs

```diff
@@ -623,26 +675,32 @@ impl<'t, 'b> Decoder<'t, 'b> {
                     args.push(Arg::F32(f32::from_bits(data)));
                 }
                 Type::BitField(range) => {
-                    let data: u64;
-
-                    match range.end {
-                        0..=8 => {
+                    let mut data: u64;
+                    let lowest_byte = range.start / 8;
+                    // -1 because `range` is range-exclusive
+                    let highest_byte = (range.end - 1) / 8;
+                    let truncated_sz = highest_byte - lowest_byte + 1; // in octets
+
+                    match truncated_sz {
+                        1 => {
                             data = self.bytes.read_u8()? as u64;
                         }
-                        0..=16 => {
+                        2 => {
                             data = self.bytes.read_u16::<LE>()? as u64;
                         }
-                        0..=24 => {
+                        3 => {
                             data = self.bytes.read_u24::<LE>()? as u64;
                         }
-                        0..=32 => {
+                        5 => {
```

Comment:
shouldn't this be `4` instead of `5`? The largest range is `0..32`; that gives `lowest_byte = 0`; `highest_byte = 3`; and `truncated_sz = 4`

Reply:
you're right– fixed and added a test to catch any future regressions